spyder/plugins/projects/tests/test_plugin.py

```diff
@@ -175,18 +175,52 @@ def test_open_project_uses_visible_config(projects, tmpdir, value):
     assert projects.dockwidget.isVisible() == value
 
 
-def test_set_get_project_filenames_when_closing(create_projects, tmpdir):
+def test_set_get_project_filenames_when_closing_no_files(create_projects,
+                                                         tmpdir):
```

Comment:
I don't understand how this new test checks a regression for issue #10045

Reply:
This is to check the old unwanted behavior of the `set_recent_files` where the method changed the `opened_files` list (because the old method modified the parameter as explained above) making the old tests pass. If you try to run this test with the old implementation of the `set_recent_files` method it fails.

Reply:
But is not the issue related to using relative paths? Or what's the real problem here?

Reply:
Yes, the main issue is related to the root path of a project being in a different mount path/drive than the files currently open. So, for example working on Windows, the project is on `C` but the files are on `D`, so no relative path exist between the project root path and the file. (that way I added the `try-except`).

However, while doing the fix, I discover that the tests which were supposed to check that the recent files elements worked on projects were kind of faulty, partially because of the way `set_recent_files` was implemented (the variable `opened_files` use to do the assert in the test, was modified by the `set_recent_files` method itself).

Reply:
Ok, thanks @dalthviz for the thorough explanation!